src/journal-remote/journal-upload.c

```diff
@@ -520,9 +526,53 @@ static int perform_upload(Uploader *u) {
+                        bool found = false;
+                        hcode = curl_easy_header(u->easy, "Accept-Encoding", 0, CURLH_HEADER, -1, &encoding);
+                        if (hcode == CURLHE_OK && encoding && encoding->value) {
+                                while(encoding->value != NULL && !found) {
+                                        _cleanup_free_ char *encVal = NULL;
+                                        _cleanup_free_ char *a = NULL;
+
+                                        if (extract_first_word((const char **)&encoding->value, &encVal, ",", 0) < 0)
+                                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to extract Accept-Encoding value");
+                                        char *t = strchr(encVal, ';');
+                                        if (t) {
+                                                a = strndup(encVal, t-encVal);
+                                                if (!a)
+                                                        return log_error_errno(SYNTHETIC_ERRNO(ENOMEM), "Failed to extract encoding from Accept-Encoding header");
+                                        } else
+                                                a = strdup(encVal);
+                                        algorithm = compression_lowercase_from_string(a);
+                                        STRV_FOREACH(c, arg_compression)
+                                                if ((*c)->algorithm == algorithm) {
+                                                        _cleanup_free_ char *header = NULL;
+                                                        struct curl_slist *headers = NULL;
+
+                                                        u->compression = *c;
+                                                        header = strjoin("Content-Encoding: ", compression_lowercase_to_string(u->compression->algorithm));
+                                                        if (!header)
+                                                                return log_oom();
+                                                        headers = curl_slist_append(NULL, header);
+                                                        if (!headers)
+                                                                return log_oom();
+                                                        headers->next = u->header->next;
+                                                        u->header->next = NULL;
+                                                        curl_slist_free_all(u->header->next);
+                                                        u->header = headers;
+                                                        easy_setopt(u->easy, CURLOPT_HTTPHEADER, u->header, LOG_ERR, return -EXFULL);
+                                                        found = true;
+                                                        break;
```

Comment:
way too much indentation, we want much shallowe indentation.

for example you can invert the `if` check above and do a `continue`, and would already save one level. but please, that'd still be too much indentation